version/derive/src/lib.rs

```diff
@@ -242,25 +242,6 @@ impl Parse for DeclareVersionedArgs {
 fn impl_decode_versioned(enum_name: &Ident) -> proc_macro2::TokenStream {
     quote! (
         impl iroha_version::scale::DecodeVersioned for #enum_name {
-            fn decode_versioned(input: &[u8]) -> iroha_version::error::Result<Self> {
-                use iroha_version::{error::Error, Version, UnsupportedVersion, RawVersioned};
-                use parity_scale_codec::Decode;
-
-                if let Some(version) = input.first() {
-                    if Self::supported_versions().contains(version) {
-                        let mut input = input.clone();
-                        Ok(Self::decode(&mut input)?)
-                    } else {
-                        Err(Error::UnsupportedVersion(Box::new(UnsupportedVersion::new(
-                            *version,
-                            RawVersioned::ScaleBytes(input.to_vec())
-                        ))))
-                    }
-                } else {
-                    Err(Error::NotVersioned)
-                }
-            }
-
             fn decode_all_versioned(input: &[u8]) -> iroha_version::error::Result<Self> {
```

Comment:
why does the implementation of `decode_all_versioned` use `::decode`?

Reply:
I thought I wouldn't change the implementation unnecessarily. But I've changed it to use decode_all now.